### ISSUE TITLE
[Package Issue]: JetBrains.IntelliJIDEA.Ultimate

### ISSUE THREAD

**Issue by khaliddermoumi** [microsoft/winget-pkgs#112662]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

winget tries to upgrade the package JetBrains.IntelliJIDEA.Ultimate 2023.1.4 with the same version any time you run upgrade.

### Steps to reproduce

The initial installation of version 2023.1.4 runs successfully. This is what the "installed" list looks like after setup:
```powershell
PS C:\Windows\System32> winget list --source winget
Name                                                    Id                                                  Version            Available
----------------------------------------------------------------------------------------------------------------------------------------
IntelliJ IDEA 2023.1.4                                  JetBrains.IntelliJIDEA.Ultimate                     > 2023.1           2023.1.4
```
But when you run "winget upgrade" afterwards, the same version is found again as an upgrade. It is then tried to run the upgrade, which fails.

### Actual behavior

winget tries to upgrade with the same package that is already installed. (This may be because the listed version is wrong)
The upgrade then fails, as the installer finds the existing installation, and exits the installation.

### Expected behavior

winget should not try to upgrade the package with the same version already installed.

### Environment

```raw
PS C:\Windows\System32> winget --info
Windows Package Manager v1.5.1572
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.19045.3208
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.20.1572.0
```

### Screenshots and Logs

n/a

**Comment by quasi-modus** [comment]
It would be great if this could be fixed permanently. This is a recurring issue and happened in the past.

**Comment by khaliddermoumi** [comment]
BTW: since the IntelliJ package has a poor "silent" setup (still requiring interaction), and is often buggy, I switched to use the "pin" feature, so my installation does not get destroyed by the package. Better a manual update than setting up an IDE repeatedly.

**Comment by treysis** [comment]
I think the problem is that IntelliJIDEA is now using PackageVersion as version number and also puts theses in the DisplayVersion under ARP, but some older manifests still have the build number listed as Version number. The build numbers numbering scheme is of course incompatible with the semantic versioning scheme and thus WinGet assumes something wrong.

Unfortunately, I assume the solution is to either purge all old(er) manifests (i.e. <=2023.1) or strip them of the "DisplayVersion" tag.

**Comment by treysis** [comment]
So, as I had the same problem with the `Community Edition`, I issued a pull request to fix the "latest" problematic manifest. Indeed, now the previous manifest to this one - that also still uses `DisplayVersion` is the problematic one.

I am not sure though if it would help to add the `DisplayVersion` to the current/latest manifest.

**Comment by lloydjatkinson** [comment]
Just came across this - is this still an issue?

**Comment by treysis** [comment]
I don't think it is. JetBrains encourages using their own Toolbox app to manage updates. I've been using this now. Not sure how an update through winget would look.
